packages/kit/src/module/define.ts

```diff
@@ -143,10 +143,12 @@ function nuxt2Shims (nuxt: Nuxt) {
         templateVars: templates.templateVars
       }
     }
+    const promises: Promise<void>[] = [];
     for await (const template of virtualTemplates) {
       const contents = await compileTemplate({ ...template, src: '' }, context)
       await fsp.mkdir(dirname(template.dst), { recursive: true })
```

Comment:
We can probably place this in another Promise.all that would run before the Promise.all(promises) if we want to really speed it up.